### ISSUE TITLE
Java 9 compatibility

### ISSUE THREAD

**Issue by eightysteele** [Macuyiko/minecraft-python#17]
Hello! 

I'm trying to install this plugin (f2ecfd2) with Spigot (1.12.2) on Java (9.0.1) with OS X (10.11.6)—but it fails with `Error occurred while enabling MinecraftPyServer v0.6 (Is it up to date?)`.

To reproduce, I exported the plugin JAR from the Eclipse project into the Spigot plugins folder, then started Spigot via `java -Xms512M -Xmx1024M -XX:MaxPermSize=128M -jar spigot-1.12.2.jar`.

I'm wondering if this is happening because of Java 9 vs Java 8—any thoughts would be great. :)

Here is the console output with the error: 

```bash
[21:20:14 INFO]: [MinecraftPyServer] Enabling MinecraftPyServer v0.6
[21:20:14 INFO]: [MinecraftPyServer] Loading MinecraftPyServerPlugin
[21:20:14 WARN]: [MinecraftPyServer] Unpacking: lib-common/java_websocket.jar
[21:20:14 WARN]: [MinecraftPyServer] Unpacking: lib-common/jython-standalone-2.7.1.jar
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/WebSocketMain.swf
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ace.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-beautify.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-chromevox.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-elastic_tabstops_lite.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-emmet.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-error_marker.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-keybinding_menu.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-language_tools.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-linking.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-modelist.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-old_ie.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-searchbox.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-settings_menu.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-spellcheck.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-split.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-static_highlight.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-statusbar.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-textarea.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-themelist.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/ext-whitespace.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/keybinding-emacs.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/keybinding-vim.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/mode-plain_text.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/mode-python.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/mode-text.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/ace/theme-twilight.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/bower.json
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/index.html
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/jq-console-master/.gitignore
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/jq-console-master/Cakefile
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/jq-console-master/css/ansi.css
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/jq-console-master/jqconsole.jquery.json
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/jq-console-master/jqconsole.min.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/jq-console-master/karma.conf.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/jq-console-master/lib/jqconsole.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/jq-console-master/package.json
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/jq-console-master/src/jqconsole.coffee
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/jquery.min.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/minecraftclient.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/style.css
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/swfobject.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: lib-http/web_socket.js
[21:20:15 WARN]: [MinecraftPyServer] Unpacking: python/mcapi.py
[21:20:15 WARN]: [MinecraftPyServer] Adding: file:/Users/eighty/Downloads/spigot/lib-common/java_websocket.jar
[21:20:15 ERROR]: Error occurred while enabling MinecraftPyServer v0.6 (Is it up to date?)
java.lang.ClassCastException: java.base/jdk.internal.loader.ClassLoaders$AppClassLoader cannot be cast to java.base/java.net.URLClassLoader
	at com.macuyiko.minecraftpyserver.MinecraftPyServerUtils.addURL(MinecraftPyServerUtils.java:70) ~[?:?]
	at com.macuyiko.minecraftpyserver.MinecraftPyServerUtils.setup(MinecraftPyServerUtils.java:29) ~[?:?]
	at com.macuyiko.minecraftpyserver.MinecraftPyServerPlugin.onEnable(MinecraftPyServerPlugin.java:26) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[spigot-1.12.2.jar:git-Spigot-b66ad9e-b5a10a9]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:337) [spigot-1.12.2.jar:git-Spigot-b66ad9e-b5a10a9]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:403) [spigot-1.12.2.jar:git-Spigot-b66ad9e-b5a10a9]
	at org.bukkit.craftbukkit.v1_12_R1.CraftServer.enablePlugin(CraftServer.java:381) [spigot-1.12.2.jar:git-Spigot-b66ad9e-b5a10a9]
	at org.bukkit.craftbukkit.v1_12_R1.CraftServer.enablePlugins(CraftServer.java:330) [spigot-1.12.2.jar:git-Spigot-b66ad9e-b5a10a9]
	at net.minecraft.server.v1_12_R1.MinecraftServer.t(MinecraftServer.java:422) [spigot-1.12.2.jar:git-Spigot-b66ad9e-b5a10a9]
	at net.minecraft.server.v1_12_R1.MinecraftServer.l(MinecraftServer.java:383) [spigot-1.12.2.jar:git-Spigot-b66ad9e-b5a10a9]
	at net.minecraft.server.v1_12_R1.MinecraftServer.a(MinecraftServer.java:338) [spigot-1.12.2.jar:git-Spigot-b66ad9e-b5a10a9]
	at net.minecraft.server.v1_12_R1.DedicatedServer.init(DedicatedServer.java:272) [spigot-1.12.2.jar:git-Spigot-b66ad9e-b5a10a9]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:545) [spigot-1.12.2.jar:git-Spigot-b66ad9e-b5a10a9]
	at java.lang.Thread.run(Thread.java:844) [?:?]
```

**Comment by Macuyiko** [comment]
Hi, thanks for the report. This is indeed due to a change between Java 8 and 9. 

Code will have to be modified to make use of the instrumentation API in Java 9 as outlined in this post: https://stackoverflow.com/a/46457506

I'll try to hack on this in the coming days and see what I can come up with.

**Comment by eightysteele** [comment]
Awesome, that sounds great. In the meantime, falling back to Java 1.8.0_151 did the trick. Thank you!

**Comment by Macuyiko** [comment]
Fixed by 6ceb6b5abe9b4731edad512b89296d174ad53144 -- still need to prepare a new release for this.
